### PR TITLE
Bugfix/multiple-containers-being-run-in-tests

### PR DESCRIPTION
**Pull request by guybortnikov**
## Description
Small bugfix to fix a race-condition where 2 containers are being run from the same test (see `test_preconditions:test_mutex_taken_twice`)

### CODE REVIEW

File: tests/test_preconditions.py [hunk 1]
```diff
@@ -20,35 +21,58 @@ def start_gprofiler(
...
-        ["-d", "1"],
+        ["-d", "1"] + extra_profiler_args,
         privileged=privileged,
         user=user,
         pid_mode=pid_mode,
     )
 
 
+@contextmanager
+def run_gprofiler_container(
```
Comment (Jongy) [inline tests/test_preconditions.py:41]: This feels like `run_gprofiler_in_container_for_one_session` without the waiting part, no? Do you want to try merging them?

File: tests/test_preconditions.py [hunk 2]
```diff
@@ -20,35 +21,58 @@ def start_gprofiler(
...
     """
-    gprofiler1 = start_gprofiler(docker_client, gprofiler_docker_image)
-    gprofiler2 = start_gprofiler(docker_client, gprofiler_docker_image)
-
-    # exits without an error
-    assert wait_for_container(gprofiler2) == (
-        "Could not acquire gProfiler's lock. Is it already running?"
-        " Try 'sudo netstat -xp | grep gprofiler' to see which process holds the lock.\n"
-    )
+    # Run the first one continuously
```
Comment (Jongy) [inline tests/test_preconditions.py:67]: This is the bug then, right? The race when the first one already exits

File: tests/test_preconditions.py [hunk 3]
```diff
@@ -20,35 +21,58 @@ def start_gprofiler(
...
+) -> Container:
+    container: Container = None
+    try:
+        container = start_gprofiler(
+            docker_client, gprofiler_docker_image, privileged, user, pid_mode, extra_profiler_args
+        )
+        yield container
+    finally:
+        if container is not None:
+            container.stop()  # Fine to be run also if already stopped
```
Comment (Jongy) [inline tests/test_preconditions.py:57]: Leaving the "remove" for a future PR?

File: tests/test_preconditions.py (hunk 2 above)
Comment (guybortnikov) [reply tests/test_preconditions.py:67]: 1. the first one might have exited before 2 have run
2. the first one didn't acquire the mutex and the second one will acquire it before (with the current solution we're waiting for log that is being logged only after the mutex successfully acquired)

File: tests/test_preconditions.py (hunk 3 above)
Comment (guybortnikov) [reply tests/test_preconditions.py:57]: yes, not related, part of https://github.com/Granulate/gprofiler/issues/600

File: tests/test_preconditions.py (hunk 1 above)
Comment (guybortnikov) [reply tests/test_preconditions.py:41]: No reason, only needed here, all the other tests are waiting for the container to exit, this is only needed when running in continuously

Comment (Jongy) [reply tests/test_preconditions.py:41]: Ok

**Review by Jongy — APPROVED** [review]

**Pull request merged**